### PR TITLE
Add possibility to push changed notes while uploading a new update

### PR DESCRIPTION
**Pull request by LChernigovskaya**
According to https://youtrack.jetbrains.com/issue/MP-2369

The notes for IntelliJ plugins will be ignored on the plugin repository side

### CODE REVIEW

File: rest/src/main/kotlin/org/jetbrains/intellij/pluginRepository/pluginRepositoryRest.kt [hunk 1]
```diff
@@ -118,12 +118,12 @@ class PluginRepositoryInstance constructor(val siteUrl: String, private val toke
             .build()
             .create(PluginRepositoryService::class.java)
 
-    fun uploadPlugin(pluginId: Int, file: File, channel: String? = null) {
-        uploadPluginInternal(file, pluginId = pluginId, channel = channel)
+    fun uploadPlugin(pluginId: Int, file: File, channel: String? = null, notes: String? = null) {
```
Comment (chashnikov) [inline rest/src/main/kotlin/org/jetbrains/intellij/pluginRepository/pluginRepositoryRest.kt]: I think adding a parameter with default value will break binary compatibility unless `@JvmOverloads` annotation is used. Could you please check this?

Comment (LChernigovskaya) [reply rest/src/main/kotlin/org/jetbrains/intellij/pluginRepository/pluginRepositoryRest.kt]: Yes, that's right, thanks. I've added this annotation.

File: cli/src/main/kotlin/org/jetbrains/intellij/pluginRepository/client.kt [hunk 2]
```diff
@@ -94,6 +94,9 @@ class Client {
 
         @set:Argument("family", description = "Plugin's family")
         var family: String? = "intellij"
+
+        @set:Argument(description = "Update's notes")
```
Comment (chashnikov) [inline cli/src/main/kotlin/org/jetbrains/intellij/pluginRepository/client.kt]: The description is not quite clear for me. How should I specify multi-line text in it? Can I use html tags? How this value is related to `change-notes` tag in `plugin.xml`?

Comment (LChernigovskaya) [reply cli/src/main/kotlin/org/jetbrains/intellij/pluginRepository/client.kt]: Yes, these are usual change notes, which can contain html tags. However, it's only for non-IntelliJ plugins (if I see correct only TeamCity plugins). For IntelliJ plugins such notes can be added but they will be ignored, and only notes from `plugin.xml` will be used.

Comment (chashnikov) [reply cli/src/main/kotlin/org/jetbrains/intellij/pluginRepository/client.kt]: Thank you for the explanation. However the current text
> Notes for non-IntelliJ plugins, HTML tags are allowed

is still not quite clear. Firstly it's not evident that "Notes" means "Change notes". And it's not clear what "non-IntelliJ" means. Will it work for WebStorm plugins, for example? I suggest to change it to something like
> Change notes (may include HTML tags). The argument is ignored when uploading updates to IntelliJ-based IDEs

Comment (LChernigovskaya) [reply cli/src/main/kotlin/org/jetbrains/intellij/pluginRepository/client.kt]: Thanks a lot!

**Pull request merged**